### DISCUSSION TITLE
Membership Application

### DISCUSSION THREAD

**Discussion by Xunzhuo** [kiali/kiali#4730] [Governance]
Hi forks!

I am a cloud native developer working at Tencent Cloud, the one of the cloud providers in China. My team, Tencent Cloud Mesh Team, is working around Istio and putting it into large scale production, which is running the multi-Cluster and multi-Control-Plane Istio in Tencent Cloud. 

After months of contributions, I have made about 30 PRs, a few issues reports, issues assignments, code reviews, discussions to kiali community. 

So now, I would like to take more responsibilities for applying as a maintainer, putting more efforts to Kiali around multicluster, performance optimization, large scale, integration with istio telemetry api, helping review PRs and codes, and also I would like to work hard to advocate kiali projects in China, let more and more Chinese Developers to use and contribute to Kiali and gather more ideas about kiali and help to make it better for a long run :)

Thanks:)

**Comment by lucasponce** [comment]
Hi, @Xunzhuo thanks for your application.
I think the next step would be to ask @kiali/maintainers to provide feedback about your request.

**Comment by lucasponce** [comment]
I'm +1 to include @Xunzhuo as a maintainer.
I think that will be positive for the community.

**Comment by leandroberetta** [comment]
+1 from my side.

He has been quite active in the community, proposing good ideas and contributing code too. I agree that will be very positive!.

**Comment by israel-hdez** [comment]
+1 as long as the hard requirements of https://github.com/kiali/kiali/blob/master/GOVERNANCE.md are properly covered. I think it's these:

> commitment to the project
> * participate in discussions, contributions, code reviews for 3 months or more,
> * perform code reviews for 10 non-trivial pull requests,
> * contribute 10 non-trivial pull requests and have them merged,

IMO, involvement in reviewing other people PRs  is the only uncovered item in that list.

**Reply by jshaughn** [reply]
Yes, I think performing reviews is missing.   I am happy to have his activity, and he is very quick to take feedback.  I think the past PRs have required a fair amount of review, I'd like to see that reduced a bit.  Also, the PRs often lack an issue or up front discussion, and they don't usually address current backlog.  With merge ability should also come more adherence to the process of issue/grooming/backlog/assignment/qe.

**Reply by Xunzhuo** [reply]
Thanks for your comments and suggestions @jshaughn, I will work hard to make myself better followed by your advice.

**Comment by Xunzhuo** [comment]
Hi @lucasponce, it has been the fifth day after application, can we move the process on?

**Reply by jmazzitelli** [reply]
As per [the official Kiali project governance process](https://github.com/kiali/kiali/blob/master/GOVERNANCE.md), the following items are listed:

```
To become a Maintainer you need to demonstrate the following:

- participate in discussions, contributions, code reviews for 3 months or more,
- perform code reviews for 10 non-trivial pull requests,
- contribute 10 non-trivial pull requests and have them merged,
```

Has it been 3 months? Have you performed code review for 10 non-trivial PRs? Have you contributed/merged 10 non-trivial PRs?

**Reply by lucasponce** [reply]
Hi @Xunzhuo, I think maintainers are positive but they would like you to keep working in the contributions and especially the pull requests review.

**Reply by Xunzhuo** [reply]
Sure, I will continue to contribute and just wait for the timing that I am fully qualified for this role, then let us disscuss again :)

**Reply by lucasponce** [reply]
+1

**Comment by jmazzitelli** [comment]
I think @Xunzhuo has now attained everything needed to be a member. How do we go about making this official? He will be given "Maintainer" status on Kiali.

Do we need official votes now? I'm +1 if so.

**Reply by Xunzhuo** [reply]
Thanks @jmazzitelli! It is my honor to be one of you forks, such an amazing OSS team. Hi @lucasponce can we move this on?

**Reply by israel-hdez** [reply]
I think you can open a PR to add him to the list of maintainers in the [GOVERNANCE.md](https://github.com/kiali/kiali/blob/master/GOVERNANCE.md) file.

You should get the needed approvals. Then, merge and give him maintainer privs.

**Reply by lucasponce** [reply]
Thanks, @Xunzhuo for contributing to the project. 

As this discussion is the "formal" request, let's give the opportunity to all maintainers to comment, and if no additional comments, we'll update the flow.

**Comment by leandroberetta** [comment]
+1

Congrats @Xunzhuo !

**Comment by jshaughn** [comment]
+1, looks forward to adding @Xunzhuo !

**Comment by Xunzhuo** [comment]
Thanks all forks, it is my great honor! I have started a PR see https://github.com/kiali/kiali/pull/5012.
